### PR TITLE
SW-4868 Enable create/update of draft sites

### PR DESCRIPTION
**Pull request by karthikbtf**
- extracted out and isolated create, update logic into hooks
- leverage these hooks in the editor
- created a placeholder hook for create planting site (follow up after BE has support)
- refactored existing code a bit to remove onChange property usage
- see attached video

screencast 2024-02-14 17-11-32

### CODE REVIEW

File: src/scenes/PlantingSitesRouter/edit/editor/Zones.tsx [hunk 1]
```diff
@@ -70,49 +71,46 @@ export default function Zones({ onChange, onValidate, site }: ZonesProps): JSX.E
...
       // check for missing zone names
       const missingZoneNames = !missingZones && zones!.features.some((zone) => !zone?.properties?.name?.trim());
+      const missingData = (missingZones || missingZoneNames) && !onValidate.isSaveAndClose;
 
-      if (zonesTooSmall) {
-        snackbar.toastError(strings.SITE_ZONE_BOUNDARIES_TOO_SMALL);
-      } else if (missingZoneNames) {
-        snackbar.toastError(strings.SITE_ZONE_NAMES_MISSING);
+      if (zonesTooSmall || missingData) {
+        snackbar.toastError(zonesTooSmall ? strings.SITE_ZONE_BOUNDARIES_TOO_SMALL : strings.SITE_ZONE_NAMES_MISSING);
```
Comment (nickgraz) [inline src/scenes/PlantingSitesRouter/edit/editor/Zones.tsx:82]: I think you need to add `activeLocale` to this `useEffect`

File: src/scenes/PlantingSitesRouter/hooks/useDraftPlantingSiteCreate.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  useEffect(() => {
+    if (draftResult?.status === 'success' && draftResult?.data && draftRequest) {
+      const id = draftResult.data;
+      if (redirect) {
+        snackbar.toastSuccess(strings.PLANTING_SITE_SAVED);
+        history.push(APP_PATHS.PLANTING_SITES_DRAFT_VIEW.replace(':plantingSiteId', `${id}`));
+      } else {
+        setCreatedDraft({
+          ...draftRequest,
+          draft: { ..._.cloneDeep(draftRequest.draft), id },
```
Comment (nickgraz) [inline src/scenes/PlantingSitesRouter/hooks/useDraftPlantingSiteCreate.ts:75]: Is `cloneDeep` needed if we're not mutating `createdDraft` directly?

**Review by nickgraz — APPROVED** [review]
Looks good to me, two small comments

File: src/scenes/PlantingSitesRouter/hooks/useDraftPlantingSiteCreate.ts (hunk 2 above)
Comment (karthikbtf) [reply src/scenes/PlantingSitesRouter/hooks/useDraftPlantingSiteCreate.ts:75]: probably not, i've been a bit paranoid about this.. will remove it, thanks

**Pull request merged**